tests/unit/test_hyper_transformer.py

```diff
@@ -1,101 +1,676 @@
+        mock.return_value = transformer
+        ht = HyperTransformer(
+            data_type_transformers={'categorical': OneHotEncodingTransformer()}
+        )
+
+        # Run
+        next_transformer = ht._get_next_transformer('a.out', 'categorical', None)
+
+        # Assert
+        assert isinstance(next_transformer, CategoricalTransformer)
+        self.assertEqual(next_transformer.fuzzy, True)
+
+    def test__update_field_types(self):
+        """Test the ``_update_field_types`` method.
+
+        This tests that if any field types are missing in the
+        provided field_types dict, that the rest of the values
+        are filled in using the data types for the dtype.
+
+        Setup:
+            - field_types will only define a few of the fields.
+
+        Input:
+            - A DataFrame of various types.
+
+        Expected behavior:
+            - field types will have values for all fields in
+            the data.
+        """
+        # Setup
+        ht = HyperTransformer(field_types={'a': 'numerical', 'b': 'categorical'})
         data = pd.DataFrame({
-            'int': [1, 2, None],
-            'float': [1.0, 2.0, None],
-            'object': ['foo', 'bar', None],
-            'category': [1, 2, None],
-            'bool': [True, False, None],
-            'datetime': pd.to_datetime(['1965-05-23', None, '1997-10-17']),
+            'a': [1, 2, 3],
+            'b': ['category1', 'category2', 'category3'],
+            'c': [True, False, True],
+            'd': [1.0, 2.0, 3.0]
         })
-        data['category'] = data['category'].astype('category')
-        result = hp._analyze(data)
 
-        # Asserts
-        assert isinstance(result, dict)
-        assert set(result.keys()) == {'int', 'float', 'object', 'category', 'bool', 'datetime'}
-
-        assert isinstance(result['int'], NumericalTransformer)
-        assert isinstance(result['float'], NumericalTransformer)
-        assert isinstance(result['object'], OneHotEncodingTransformer)
-        assert isinstance(result['category'], OneHotEncodingTransformer)
-        assert isinstance(result['bool'], BooleanTransformer)
-        assert isinstance(result['datetime'], DatetimeTransformer)
-
-    def test__analyze_invalid_dtype(self):
-        """Test _analyze when a list of dtypes containing an invalid dtype is passed."""
+        # Run
+        ht._update_field_types(data)
+
+        # Assert
+        expected = {'a': 'numerical', 'b': 'categorical', 'c': 'boolean', 'd': 'float'}
+        self.assertEqual(ht.field_types, expected)
+
+    @patch('rdt.hyper_transformer.load_transformer')
+    def test__fit_field_transformer(self, load_transformer_mock):
+        """Test the ``_fit_field_transformer`` method.
+
+        This tests that the ``_fit_field_transformer`` behaves as expected.
+        It should fit the transformer it is provided, loops through its
+        outputs, transform the data if the output is not ML ready and call
```

Comment:
Just a question - How do you determine if the output is ML ready?

Reply:
Good question! We check the `_transform_output_types` list. I will update this documentation to show that in the other PR. I'm trying to keep this PR strictly about tests.